### PR TITLE
Xdo

### PR DESCRIPTION
**Pull request by Liupold**
Seems like a better way.

### CODE REVIEW

**Comment by SeerLite** [comment]
Tested it on KDE and Xfce, all seems to work fine on my end.

**Review by SeerLite — DISMISSED** [review]

File: pidswallow [hunk 1]
```diff
@@ -65,59 +72,60 @@ swallow() {
...
         [ -z "$ppid" ] && return 1 # nothing swallowable.
-        [ "$verbose" -eq 1 ] && echo "swallow - $pname($ppid) detected as swallowable"
-        pwid="$(xdotool search --limit 1 --onlyvisible --pid "$ppid")"
+        vecho "swallow - $pname($ppid) detected as swallowable"
+        pwid="$(xdo id -p "$ppid" | sed 1q)"
         [ -z "$pwid" ] && return 4
-        [ "$verbose" -eq 1 ] && echo "swallow - $cwid swallowing $pwid using $swallow_command"
-        echo "$swallow_command" | \
+        vecho "swallow - $cwid swallowing $pwid using $swallow_command"
+        xdo activate "$pwid"; echo "$swallow_command" | \
```
Comment (SeerLite) [inline pidswallow]: This one is giving me problems. Why activate before swallowing? It messes up some fullscreen rules (re-focusing takes child program out of fullscreen).

**Review by SeerLite — CHANGES_REQUESTED** [review]

File: pidswallow (hunk 1 above)
Comment (Liupold) [reply pidswallow:96]: It's got here by accident. Changing.

Comment (Liupold) [reply pidswallow]: Pls push your change.

Comment (Liupold) [reply pidswallow]: It's to maintain the correct shape in floating WM. (those with dynamic tiling). Eg: bspwm. Sorry I really forgot about it

Comment (Liupold) [reply pidswallow]: I think it is better to have pre glue hook before swallowing. As some wm don't manage the window after unmaping.

Comment (Liupold) [reply pidswallow]: Fixed it, it was doing nothing. 😖

**Comment by Liupold** [comment]
Merging this, (There are a lot of branch to track).

**Pull request merged**